Lib/smtplib.py

```diff
@@ -953,14 +957,16 @@ def send_message(self, msg, from_addr=None, to_addrs=None,
                     "One or more source or delivery addresses require"
                     " internationalized email support, but the server"
                     " does not advertise the required SMTPUTF8 capability")
-            international = True
-        with io.BytesIO() as bytesmsg:
-            if international:
-                g = email.generator.BytesGenerator(
-                    bytesmsg, policy=msg.policy.clone(utf8=True))
-                mail_options += ['SMTPUTF8', 'BODY=8BITMIME']
+            policy = policy.clone(utf8=True)
+            mail_options += ['SMTPUTF8', 'BODY=8BITMIME']
+        if policy.cte_type == '8bit':
+            if self.has_extn('8bitmime'):
+                if 'BODY=8BITMIME' not in mail_options:
+                    mail_options.append('BODY=8BITMIME')
```

Comment:
Looks like we could end up appending BODY=8BITMIME twice (once in the except, and then again here).  Probably want to keep the international flag and append SMTPUTF8 iff it is true.

Reply:
I called the boolean differently since this is about an 8-bit body and not about internationalized mail.